### ISSUE TITLE
MME Crash on assertions

### ISSUE THREAD

**Issue by bmurithi** [magma/magma#418]
MME repeatedly crashes with these errors.
As a result, the cell keeps going online/offline.

```
Jul 09 15:30:41 lte mme[14894]: Directoryd RPC failed with code 2, msg: Exception calling application: <_Rendezvous of RPC that terminated with:
Jul 09 15:30:41 lte mme[14894]:         status = StatusCode.UNKNOWN
Jul 09 15:30:41 lte mme[14894]:         details = "Error finding location record: No record for query"
Jul 09 15:30:41 lte mme[14894]:         debug_error_string = "{"created":"@1562675441.688892940","description":"Error received from peer","file":"src/core/lib/surface/call.cc","file_line":1017,"grpc_message":"Error finding location record: No record for query","grpc_status":2}"
Jul 09 15:30:41 lte mme[14894]: >
Jul 09 15:30:46 lte mme[14894]: Assertion (0) failed!
Jul 09 15:30:46 lte mme[14894]: In itti_receive_msg() /home/admin/magma/lte/gateway/c/oai/lib/itti/intertask_interface.c:666
Jul 09 15:30:46 lte mme[14894]: No message in queue for task 4 while there are 1 events and some for the messages queue!
Jul 09 15:30:46 lte mme[14894]:  Will Intentionaly raise SEGFAULT to be catched by GDB!
Jul 09 15:30:46 lte mme[14894]: Obtained 5 stack frames.
Jul 09 15:30:46 lte mme[14894]: /usr/local/bin/mme(display_backtrace+0x20) [0x5563f522957d]
Jul 09 15:30:46 lte mme[14894]: /usr/local/bin/mme(itti_receive_msg+0x45a) [0x5563f54ea0ba]
Jul 09 15:30:46 lte mme[14894]: /usr/local/bin/mme(mme_app_thread+0xf2) [0x5563f52bf940]
Jul 09 15:30:46 lte mme[14894]: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f6f8828c494]
Jul 09 15:30:46 lte mme[14894]: /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f6f84ffbacf]
Jul 09 15:30:55 lte systemd[1]: magma@mme.service: Main process exited, code=killed, status=11/SEGV
Jul 09 15:30:55 lte systemd[1]: magma@mme.service: Unit entered failed state.
Jul 09 15:30:55 lte systemd[1]: magma@mme.service: Failed with result 'signal'.
Jul 09 15:31:00 lte systemd[1]: magma@mme.service: Service hold-off time over, scheduling restart.
Jul 09 15:31:00 lte systemd[1]: Stopped Magma OAI MME service.
Jul 09 15:31:01 lte systemd[1]: Starting Magma OAI MME service...
Jul 09 15:31:04 lte mme[15327]: ERROR:root:Error retrieving config for mme, key not found: use_stateless
Jul 09 15:31:04 lte mme[15327]: WARNING:root:Service (spgw) missing in mconfig
Jul 09 15:31:05 lte systemd[1]: Started Magma OAI MME service.
Jul 09 15:31:05 lte mme[15368]: [/home/admin/magma/orc8r/gateway/c/common/config/MConfigLoader.cpp:65] Couldn't parse pipelined config
Jul 09 15:31:05 lte mme[15368]: [INFO] Unable to load mconfig for mme, S6a relay is disabled
Jul 09 15:31:05 lte mme[15368]: [/home/admin/magma/orc8r/gateway/c/common/config/MConfigLoader.cpp:65] Couldn't parse pipelined config
Jul 09 15:31:05 lte mme[15368]: [INFO] Unable to load mconfig for mme, cloud subscriberdb is disabled
Jul 09 15:31:05 lte mme[15368]: Initializing OAI Logging
Jul 09 15:31:05 lte mme[15368]: Initializing OAI Logging to syslog
Jul 09 15:31:05 lte MME[15368]: 000000 Tue Jul  9 15:31:05 2019 7F6BD122D400 INFO  UTIL   common/log.c                    :0873    Initializing OAI logging Done



Jul 09 15:31:47 lte mme[15368]: Assertion (0) failed!
Jul 09 15:31:47 lte mme[15368]: In itti_receive_msg() /home/admin/magma/lte/gateway/c/oai/lib/itti/intertask_interface.c:666
Jul 09 15:31:47 lte mme[15368]: No message in queue for task 4 while there are 1 events and some for the messages queue!
Jul 09 15:31:47 lte mme[15368]:  Will Intentionaly raise SEGFAULT to be catched by GDB!
Jul 09 15:31:47 lte mme[15368]: Obtained 5 stack frames.
Jul 09 15:31:47 lte mme[15368]: /usr/local/bin/mme(display_backtrace+0x20) [0x559d6249157d]
Jul 09 15:31:47 lte mme[15368]: /usr/local/bin/mme(itti_receive_msg+0x45a) [0x559d627520ba]
Jul 09 15:31:47 lte mme[15368]: /usr/local/bin/mme(mme_app_thread+0xf2) [0x559d62527940]
Jul 09 15:31:47 lte mme[15368]: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f6bcfd68494]
Jul 09 15:31:47 lte mme[15368]: /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f6bccad7acf]
Jul 09 15:31:56 lte systemd[1]: magma@mme.service: Main process exited, code=killed, status=11/SEGV
Jul 09 15:31:56 lte systemd[1]: magma@mme.service: Unit entered failed state.
Jul 09 15:31:56 lte systemd[1]: magma@mme.service: Failed with result 'signal'.
Jul 09 15:32:01 lte systemd[1]: magma@mme.service: Service hold-off time over, scheduling restart.
Jul 09 15:32:01 lte systemd[1]: Stopped Magma OAI MME service.
Jul 09 15:32:01 lte systemd[1]: Starting Magma OAI MME service...
Jul 09 15:32:05 lte mme[15626]: ERROR:root:Error retrieving config for mme, key not found: use_stateless
Jul 09 15:32:05 lte mme[15626]: WARNING:root:Service (spgw) missing in mconfig
Jul 09 15:32:06 lte systemd[1]: Started Magma OAI MME service.
Jul 09 15:32:06 lte mme[15651]: [/home/admin/magma/orc8r/gateway/c/common/config/MConfigLoader.cpp:65] Couldn't parse pipelined config
Jul 09 15:32:06 lte mme[15651]: [INFO] Unable to load mconfig for mme, S6a relay is disabled
Jul 09 15:32:06 lte mme[15651]: [/home/admin/magma/orc8r/gateway/c/common/config/MConfigLoader.cpp:65] Couldn't parse pipelined config
Jul 09 15:32:06 lte mme[15651]: [INFO] Unable to load mconfig for mme, cloud subscriberdb is disabled
Jul 09 15:32:06 lte mme[15651]: Initializing OAI Logging
Jul 09 15:32:06 lte mme[15651]: Initializing OAI Logging to syslog
Jul 09 15:32:06 lte MME[15651]: 000000 Tue Jul  9 15:32:06 2019 7F3FC172A400 INFO  UTIL   common/log.c                    :0873    Initializing OAI logging Done
Jul 09 15:32:06 lte mme[15651]: Initializing shared logging
Jul 09 15:32:06 lte mme[15651]: Initializing shared logging Done
```

On the same orchestrator, a different network is stable.

# Tested Magma Gateway Versions

- 0.3.74
- 0.3.73
- 0.3.72

**Comment by bmurithi** [comment]
Feels like there should be a tie-in between gateway version/orchestrator versions.

**Comment by ssanadhya** [comment]
Fixed in version 0.3.75
